crowbar_framework/app/controllers/machines_controller.rb

```diff
@@ -71,39 +71,32 @@ def show
   add_help(:role, [:id])
   def role
     @machine.intended_role = params[:role]
-    @machine.save
 
     respond_to do |format|
-      format.json { head :ok }
+      if @machine.save
+        format.json { head :ok }
+      else
+        format.json do
+          render json: { error: "#{I18n.t('cannot_save_node', scope:'error')} #{@machine.name}" },
+            status: :unprocessable_entity
```

Comment:
Style/AlignHash: Align the elements of a hash literal if they span more than one line.